posthog/session_recordings/queries/session_recording_list_from_filters.py

```diff
@@ -485,7 +486,7 @@ def _having_predicates(self) -> ast.Expr:
 
         if event_names:
             return ast.Call(
-                name="hasAll",
+                name="hasAll" if self._filter._operand == "AND" else "hasAny",
```

Comment:
I'm assuming this is an optimisation. We cannot rely on every event being in the array when using OR conditions